actionpack/test/controller/render_test.rb

```diff
@@ -368,6 +379,18 @@ def test_expires_now_with_cache_control_headers
     assert_match(/no-transform/, @response.headers["Cache-Control"])
   end
 
+  def test_expires_now_with_conflicting_cache_control_headers
+    get :conditional_hello_with_expires_and_confliciting_cache_control_headers
+    assert_match(/no-cache/, @response.headers["Cache-Control"])
+    refute_match(/must-revalidate/, @response.headers["Cache-Control"])
```

Comment:
Swapped for equality assertions in https://github.com/rails/rails/pull/30367/commits/2b4c547ea803e3de8684b7cc3665b9a1cd0e6d3a.